### ISSUE TITLE
Incorrect agglomeration and breakage BD

### ISSUE THREAD

**Issue by wuyuanyi135** [wuyuanyi135/crm#8]
The kernel unit (m^-3 * s^-1) of B and D were learnt from the 2020 paper. They were incorrectly taken as “per volume of the crystallizer per second”. This caused impossible magnitude of D that required very small time step.

After reviewing the definition, this will be the correct way to understand the unit and formula of the kernel.

* During BRK and AGG, the number density should be converted into particle volume coordination by converting the particle sizes into the corresponding volume. Hence, the unit of the converted number density changed from #/um to #/um^3.
* The count involved in the AGG or BRK is kernel * number density of size 1 * number density of size 2 * dV^2 (use AGG as example). The dV^2 is to convert the number density into number. dV is not the constant obtained from the product of the dimension intervals! dV is the integral variable of particle volume. However, we already have particle count here, so ours is kernel * count of size 1 * count of size 2. Hence, no need to worry about the dV here. 
* The reason why kernel has the volume unit is that the LHS of the PBE is about dn/dt, which has the unit of # * m^-3 * s^-1. Now this kernel unit matches it. 
In conclusion, the meaning of the kernel’s unit is “For the particles of volume Vp, if two classes of particles whose volumes are Vp1 and Vp2, where Vp1 + Vp2 == Vp, how much particle will be involved in the AGG process, per second, **per volume of the particle** (because of the number density in volume coordination)?

So, we should just multiply the kernel * ... with the agglomerated particle’s particle volume to get the count per second involved in this process.

**Comment by wuyuanyi135** [comment]
Note that the AGG and BRK are all considering the total population in the system, not per volume property. i.e., the concentrate or dilute solids concentration does not affect the result of AGG or BRK unless the kernel take these factors into consideration. 

In our calculation, volume of the crystallizer is still needed to compute the number of population. It is still possible to get some number that cause time step exception. As long as the parameter magnitude is ok, we should not worry about it too much.

**Comment by wuyuanyi135** [comment]
Multiplying the crystall volume is not the correct way! Note that the coordination is volume-based, so the number density has a unit of `#/m^3`. This m^3 will be cancelled when integral along the volume grid `\int_{V-\Delta V}^{V+\Delta V}{ndV}`, which will gives the number of crystals whose volumes are between V-Delta V and V+ Delta V. 

The problem is that this method record a table of crystal sizes and counts. There is no grid. But this table must be presenting a continuous distribution. Otherwise, the system will be consisting of a series of Dirac pulses. But what range does each row represent? For example, when a row is 1um, 10000, does it cover 1+-0.1um? or 0.5um?

Current thought: compress n before and after agglomeration
Before: the grid is implicitly created. we can use the interval as the covered range of each row. Also the # of rows may be reduced
After: for performance.
